src/reduxPromiseListener.js

```diff
@@ -66,12 +66,16 @@ export default function createListener(): PromiseListener {
           }
         }
         listeners[listenerId] = listener
-        dispatch(
-          (config.setPayload || defaultSetPayload)(
+        let action
+        if (typeof config.start === 'function') {
+          action = config.start({ payload, meta: { id } })
+        } else {
+          action = (config.setPayload || defaultSetPayload)(
```

Comment:
Just for completeness, why can't we have the `setPayload` outside of this? So that it runs on `action` after that has been set either way. Or alternatively I suppose completely remove `setPayload` in favour of using a function for `start`.

Reply:
I left that in so that we don't have to refactor everything in the dApp again. Once we have done this, we can safely remove it.